codemagic_manager/lib/src/codemagic_client.dart

```diff
@@ -141,4 +236,191 @@ class CodemagicClient {
       return ApiResponse(wasSuccessful: false, error: 'Unknown error');
     }
   }
+
+  /// Retrieves artifacts for specific [Build] by its [buildId]
+  Future<ApiResponse<Artifacts>> getArtifacts(String buildId) async {
```

Comment:
Artifacts fetching works differently, please reread the documentation. Each build includes list of artefacts and then for each artefact you can generate a public url https://github.com/codemagic-ci-cd/codemagic-docs/blob/master/content/rest-api/artifacts.md